File: math/fibonacci_number.py
class Solution:
    def fib(self, n: int) -> int:
        if n == 0 or n == 1:
            return n
        # build fibonacci sequence
        fib = []
        for i in range(n + 1):
            if i == 0 or i == 1:
                fib.append(i)
            else:
                fib.append( fib[i - 2] + fib[i - 1] )
        # add last two entries of fib (the two before n)
        return fib[n]

File: math/fibonacci_number.rb
# @param {Integer} n
# @return {Integer}
def fib(n)
    return n if n == 0 || n == 1
    # build fibonacci sequence
    fib = []
    (0..n).each_with_index do |num, i|
        if i == 0 || i == 1
            fib << num
        else
            fib << fib[i - 2] + fib[i - 1]
        end
    end
    # add last two entries of fib (the two before n)
    fib[-1]
end